### DISCUSSION TITLE
How to retrieve attribute elements

### DISCUSSION THREAD

**Discussion by th0masj0** [MicroStrategy/rest-api-playground#4] [General]
Hey, 

I‘m searching for REST API method for retrieving all attribute elements of an attribute. It seems there is no method. Thx 

Regards
Thomas

**Comment by xdumstr** [comment]
To get the elements for an attribute, you need to get the data from the data source, not only the metadata.  Take a look at GET /api/reports/{reportId}/attributes/{attributeId}/elements](https://demo.microstrategy.com/MicroStrategyLibrary/api-docs/index.html#/Reports/getAttributeElementFromReport) and [GET /api/cubes/{cubeId}/attributes/{attributeId}/elements](https://demo.microstrategy.com/MicroStrategyLibrary/api-docs/index.html#/Cubes/getCubeAttributeElement).  You can also take a look at the workflow example in [REST API Playground](https://www.postman.com/microstrategysdk/workspace/microstrategy-rest-api/folder/16131298-84db2127-16e8-4fd2-909f-b5990516f889?ctx=documentation)

**Reply by th0masj0** [reply]
Hey @xdumstr, thanks for your reply. Your recommendation requires to create a report or cube. This is very uncomfortable. It will be better to simply get a list of all attribute elements like in MicroStrategy Web and Developer, e.g. hierarchy browsing or in attribute filter selection.

**Reply by xdumstr** [reply]
@thomasjoscht,  thank you for your feedback. We will pass it to the SDK team.

**Comment by th0masj0** [comment]
Hey @elinmstr, that's it! Using [GET /api/attributes/{id}/elements](https://demo.microstrategy.com/MicroStrategyLibrary/api-docs/index.html?visibility=all#/Attributes/getAttributeElements) does work for MicroStrategy 2021 Update 7 in addition to /api/attributes/{id} for extraction of attribute forms.

PS: your hint with visibility=all was a "good to know" thing. Thx

**Reply by xiakunhou** [reply]
Note that, visibility to all includes some private APIs which not been public yet. Glade that it works though.
